### PR TITLE
Implementing signup signin

### PR DESCRIPTION
**Pull request by asavienko**

### CODE REVIEW

File: src/components/ReusableComponents/ShowResult.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Result } from "antd";
+import React from "react";
+
+const ShowResult = ({ status = "success", title, subTitle, buttonsArr }) => (
```
Comment (bohdanSnappy) [inline src/components/ReusableComponents/ShowResult.js]: 1) remove Show result component
2) use Result component directly

File: src/components/SignIn/SignIn.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    rule ? callback() : callback("Test");
+  };
+
+  render() {
+    const { getFieldDecorator } = this.props.form;
+
+    return (
+      <Form onSubmit={this.handleSubmit} {...formItemLayout}>
+        <Form.Item>
+          {getFieldDecorator("email", {
```
Comment (bohdanSnappy) [inline src/components/SignIn/SignIn.js]: override default tooltip

File: src/components/SignIn/SignIn.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    const { getFieldDecorator } = this.props.form;
+
+    return (
+      <Form onSubmit={this.handleSubmit} {...formLayout}>
+        <Form.Item>
+          {getFieldDecorator("email", {
+            rules: [{ required: true, message: "Пожалуйста введите Ваш email" }]
+          })(
+            <Input
+              prefix={<Icon type="mail" style={{ color: "rgba(0,0,0,.25)" }} />}
```
Comment (bohdanSnappy) [inline src/components/SignIn/SignIn.js]: move to reusable styled component

File: src/components/SignIn/SignIn.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            <Input
+              prefix={<Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />}
+              type="password"
+              placeholder="Пароль"
+            />
+          )}
+        </Form.Item>
+        <Form.Item>
+          <Row type="flex" justify="space-between">
+            {getFieldDecorator("remember", {
```
Comment (bohdanSnappy) [inline src/components/SignIn/SignIn.js]: "authed"

File: src/components/SignUp/SignUp.js [hunk 5]
```diff
@@ -49,13 +51,22 @@ class SignUp extends React.Component {
...
 
-      console.log("Dispatch data: ", { name, surname, email, password, phone });
+      const response = await postRequest("/users/sign-up", {
+        name,
+        surname,
+        email,
+        password,
+        phone
+      });
+      console.log(response);
```
Comment (bohdanSnappy) [inline src/components/SignUp/SignUp.js]: remove console.log

File: src/components/TopMenu/TopMenu.js [hunk 6]
```diff
@@ -1,25 +1,67 @@
...
+  process.env.PUBLIC_URL + "images/temporaryImages/lines_layout.png";
 
 const StyledMenu = styled(Menu)`
   line-height: 45px;
+  ${({ show }) => show === "false" && `visibility: hidden;`}
 `;
-
-function TopMenu() {
+const HeaderLayer = styled.div`
+  display: none;
```
Comment (bohdanSnappy) [inline src/components/TopMenu/TopMenu.js]:
```suggestion
  display: ${props => props.show === "true" ? "none" : "block"}
```

Comment (asavienko) [reply src/components/TopMenu/TopMenu.js]: After the changes, I have an error 
image

File: src/components/SignIn/SignIn.js (hunk 3 above)
Comment (asavienko) [reply src/components/SignIn/SignIn.js]: After the changes, I have an error below: 
image

Comment (asavienko) [reply src/components/SignIn/SignIn.js]: After the Icon removing, I have the same error
image

Comment (asavienko) [reply src/components/SignIn/SignIn.js]: Should I resolve this problem or just pass styles as props?

File: src/services/fetchUtils.js [hunk 7]
```diff
@@ -12,6 +12,7 @@ async function request(url, method, body) {
       headers,
       body: body ? JSON.stringify(body) : undefined
     });
+    console.log(response);
```
Comment (bohdanSnappy) [inline src/services/fetchUtils.js]: remove console.log

File: src/components/SignIn/SignIn.js [hunk 8]
```diff
@@ -19,8 +20,12 @@ class SignIn extends React.Component {
     e.preventDefault();
     const { validateFields } = this.props.form;
 
-    const onSuccess = values => {
-      console.log(values);
+    const onSuccess = async ({ remember, ...dataToSend }) => {
+      const response = await postRequest("/users/sign-in", { ...dataToSend });
+      console.log(response);
```
Comment (bohdanSnappy) [inline src/components/SignIn/SignIn.js]: remove console.log

File: src/components/SignUp/SignUp.js [hunk 9]
```diff
@@ -56,14 +56,9 @@ class SignUp extends React.Component {
...
-        .then(response => {
-          console.log("success" + JSON.stringify(response));
-        })
-        .catch(error => {
-          console.log("error" + error);
-        });
-      console.log(response);
+      });
+
+      console.log(response.err);
```
Comment (bohdanSnappy) [inline src/components/SignUp/SignUp.js]: open notification

Comment (bohdanSnappy) [reply src/components/SignUp/SignUp.js]: 1) create reusable logic for login function
2) login function should return user
3) check if user.emailVerified === true ? redirect ot main page :  redirect to email not verified page

Comment (asavienko) [reply src/components/SignUp/SignUp.js]: image
image

I have no ideas how to handle errors?!

File: api/services/users/userServices.js [hunk 10]
```diff
@@ -30,16 +30,10 @@ const authenticate = async ({ email, password }) => {
   const [user] = await collection
     .aggregate(userPipelines.authenticate(email))
     .toArray();
-  if (!user.emailVerified) throw "Email is not verified";
   if (user && bcrypt.compareSync(password, user.hash)) {
-    const { hash, emailVerified, ...userWithoutHash } = user;
-
+    const { hash, ...userWithoutHash } = user;
```
Comment (bohdanSnappy) [inline api/services/users/userServices.js]: check this logic

File: src/components/ReusableComponents/StyledInput.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import styled from "styled-components";
+import { Input } from "antd";
+
+const StylesForInput = styled(Input)`
```
Comment (bohdanSnappy) [inline src/components/ReusableComponents/StyledInput.js]: move to .styles.js file

File: src/components/ReusableComponents/StyledInput.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+  .ant-input-prefix {
+    color: rgba(0, 0, 0, 0.25);
+  }
+`;
+
+class StyledInput extends React.Component {
```
Comment (bohdanSnappy) [inline src/components/ReusableComponents/StyledInput.js]: remove this component. Reuse StylesForInput from .styles.js file

File: src/components/SignIn/SignIn.js [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+class SignIn extends React.Component {
+  handleSubmit = e => {
+    e.preventDefault();
+    const { validateFields } = this.props.form;
+
+    const onSuccess = async ({ ...dataToSend }) => {
+      const response = await postRequest("/users/sign-in", { ...dataToSend });
+
+      const history = this.props.history;
```
Comment (bohdanSnappy) [inline src/components/SignIn/SignIn.js]:
```suggestion
      const {history} = this.props;
```

File: src/components/SignUp/SignUp.js [hunk 14]
```diff
@@ -49,13 +51,28 @@ class SignUp extends React.Component {
...
-      console.log("Dispatch data: ", { name, surname, email, password, phone });
+      const response = await postRequest("/users/sign-up", {
+        name,
+        surname,
+        email,
+        password,
+        phone
+      });
+
+      const history = this.props.history;
```
Comment (bohdanSnappy) [inline src/components/SignUp/SignUp.js]: see comment above

File: src/components/TopMenu/TopMenu.js [hunk 15]
```diff
@@ -1,25 +1,68 @@
...
 import styled from "styled-components";
+import { withRouter } from "react-router";
+const employ_logo =
+  process.env.PUBLIC_URL + "images/temporaryImages/employ_logo.svg";
+const lines_layout =
+  process.env.PUBLIC_URL + "images/temporaryImages/lines_layout.png";
 
 const StyledMenu = styled(Menu)`
   line-height: 45px;
+  ${({ show }) => show === "false" && `visibility: hidden`}
```
Comment (bohdanSnappy) [inline src/components/TopMenu/TopMenu.js]:
```suggestion
  ${({ show }) => show === "false" && `visibility: hidden`}
  visibility: ${({ show }) => show === "false" ? 'hidden' : 'visible'}
```

File: src/services/signUpSignInFunction.js [hunk 16]
```diff
@@ -0,0 +1,49 @@
+import openNotification from "../components/ReusableComponents/Notification";
+import Cookies from "js-cookie";
+import { postRequest } from "./fetchUtils";
+
+const signUpSignInFunction = ({ response, history, loginData = {} }) => {
```
Comment (bohdanSnappy) [inline src/services/signUpSignInFunction.js]:
```suggestion
const signUpSignInFunction = ({ response = {}, history = {} , loginData = {} }) => {
```

File: src/services/signUpSignInFunction.js [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+  const isError = response && "err" in response;
+  const isEmailVerifiedFalse =
+    response && response.token && response.emailVerified === false;
+  const isEmailVerifiedTrue =
+    response && response.token && response.emailVerified === true;
```
Comment (bohdanSnappy) [inline src/services/signUpSignInFunction.js]: const validToken = !!response.token;

File: src/services/signUpSignInFunction.js [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+        type: "error",
+        message: "Ошибка",
+        description: response.err
+      });
+    case isEmailVerifiedFalse:
+      Cookies.set("Access-Token", response.token);
+      return history.push("/email-not-verified");
+    case isEmailVerifiedTrue:
+      Cookies.set("Access-Token", response.token);
+      return history.push("/");
```
Comment (bohdanSnappy) [inline src/services/signUpSignInFunction.js]: case validToken:
Cookies.set("Access-Token", response.token);
rerurn history.push(response.emailVerified ? "/" : "/email-not-verified")

File: api/services/users/userServices.js (hunk 10 above)
Comment (asavienko) [reply api/services/users/userServices.js]: I didn't find a way to improve this logic

**Pull request merged**